src/librustc_ty/instance.rs

```diff
@@ -36,11 +36,6 @@ pub fn resolve_instance<'tcx>(
                 let ty = substs.type_at(0);
 
                 if ty.needs_drop(tcx, param_env) {
-                    // `DropGlue` requires a monomorphic aka concrete type.
-                    if ty.needs_subst() {
-                        return None;
-                    }
-
```

Comment:
You'd need to make sure the type is something drop glue can be generated for.

Reply:
Fixed in 89f8ec0.